lib/db/schema/patch-029-30.sql

```diff
@@ -0,0 +1,23 @@
+-- Update deleteSessionToken stored procedure to delete from devices.
+CREATE PROCEDURE `deleteSessionToken_3` (
+  IN `tokenIdArg` BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM devices WHERE sessionTokenId = tokenIdArg;
```

Comment:
Sadly there's no index on `sessionTokenId`, so this would result in a full table scan of a pretty sizeable table:

```
mysql> explain DELETE FROM devices WHERE sessionTokenId = 'foobar';
+----+-------------+---------+------------+------+---------------+------+---------+------+------+----------+-------------+
| id | select_type | table   | partitions | type | possible_keys | key  | key_len | ref  | rows | filtered | Extra       |
+----+-------------+---------+------------+------+---------------+------+---------+------+------+----------+-------------+
|  1 | DELETE      | devices | NULL       | ALL  | NULL          | NULL | NULL    | NULL |    1 |   100.00 | Using where |
+----+-------------+---------+------------+------+---------------+------+---------+------+------+----------+-------------+
1 row in set (0.01 sec)
```

There *is* an index on (uid, sessionTokenId) that could be used for this, but we'd have to change the way this procedure is called in order to pass in the uid as well:

```
 explain DELETE FROM devices WHERE uid='x' AND sessionTokenId = 'foobar';
+----+-------------+---------+------------+-------+-----------------------------------+---------+---------+-------+------+----------+-------------+
| id | select_type | table   | partitions | type  | possible_keys                     | key     | key_len | ref   | rows | filtered | Extra       |
+----+-------------+---------+------------+-------+-----------------------------------+---------+---------+-------+------+----------+-------------+
|  1 | DELETE      | devices | NULL       | range | PRIMARY,UQ_devices_sessionTokenId | PRIMARY | 16      | const |    1 |   100.00 | Using where |
+----+-------------+---------+------------+-------+-----------------------------------+---------+---------+-------+------+----------+-------------+
1 row in set (0.00 sec)
```